lib/ansible/plugins/callback/json.py

```diff
@@ -100,15 +100,24 @@ def v2_playbook_on_stats(self, stats):
             s = stats.summarize(h)
             summary[h] = s
 
-        custom_stats = {}
+        custom_stats = {
+            'global': {},
+            'hosts': {}
+        }
+
         if self.get_option('show_custom_stats') and stats.custom:
-            custom_stats.update(dict((self._convert_host_to_name(k), v) for k, v in stats.custom.items()))
-            custom_stats.pop('_run', None)
+
+            for k in sorted(stats.custom.keys()):
+                if k == '_run':
+                    custom_stats['global'] = stats.custom[k]
+                    continue
+
+                custom_stats['hosts'].update({self._convert_host_to_name(k): stats.custom[k]})
 
         output = {
             'plays': self.results,
             'stats': summary,
-            'custom_stats': custom_stats,
+            'custom_stats': custom_stats
```

Comment:
Please do not remove this trailing comma, this is our preference with dicts and lists to minimize future diffs when lines are changed.

Reply:
refactor it that custom_stats have the same output as before and add a additional dict custom_stats_global